packages/block-library/src/columns/block.json

```diff
@@ -1,6 +1,7 @@
 {
 	"name": "core/columns",
 	"category": "layout",
+	"selector": ".wp-block-columns",
```

Comment:
This should be the default and only needed for blocks that don't have the generated class.

Reply:
ah, so would you prefer doing:

- `selector: true`, for blocks that use the generated class
- `selector: <some-other-selector>` for blocks that don't

Reply:
After [Greg's comment](https://github.com/WordPress/gutenberg/pull/22698#issuecomment-635268128) I'm thinking of moving this under the supports key and do the same as other experimental features, so we'd have:

```
supports: {
  __experimentalSelector: {boolean|string|object}
}
```

Reply:
> selector: true, for blocks that use the generated class

Why do we need the `selector: true`. Can't we just consider it as the default if the "selector" is not specified?

Reply:
This is not part of the block type RCF is it? @gziolo ?

Reply:
Yes I agree if no selector is specified at all we should target the block by the default class.

Reply:
As automatic selectors we should also use the block styles:
e.g: 'core/quote/large' automatically mapped to .wp-block-quote.is-style-large.

Reply:
Yes I agree if no selector is specified at all we should target the block by the default class.

Reply:
> Can't we just consider it as the default if the "selector" is not specified?

Yeah, I think we can.

My mind-frame in coming to this was that it'd be an opt-in mechanism, but it can perhaps work fine by generating the class if no selector is provided. Thinking out loud: what would be the worst that can happen if this isn't opt-in? This is what I can think of:

1. We're duplicating the class generation logic client & server-side. I think this should be fine because that's an area that I don't expect to change.

2. Class conflicts due to namespacing not matching the class name. Ex:
  - `core/media-text` block name => `.wp-block-media-text` class name.
  - `custom-library/media-text` block name => `.wp-block-media-text` class name.

I don't think third-party blocks are overriding core's class names, otherwise, this is problematic on its own. So, perhaps what we should do is to generate this class by default: `.wp-{namespace}-{block-name}` to prevent conflicts. With the exception that we don't add the namespace for core blocks, so they'll be `.wp-{block-name}` instead.

Reply:
> As automatic selectors we should also use the block styles:
e.g: 'core/quote/large' automatically mapped to .wp-block-quote.is-style-large.

@jorgefilipecosta I think adding support for style variations merits its own PR / discussion. The first step would be to look at them as "sets of style attributes" as suggested at https://github.com/WordPress/gutenberg/issues/20331Perhaps after that experiment we end up doing some things (like inlining those values) that remove the need of targeting the existing classes.